contracts/chain-adapters/ForwarderBase.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title ForwarderBase
+ * @notice Base contract deployed some network between L1 and the network housing a spoke pool contract. If a message is sent
```

Comment:
Might be helpful to note that messages received on this contract are expected to be sent from the `rerouter_adapter` or w.e. we decide to name it